nx/source/services/set.c

```diff
@@ -686,3 +686,69 @@ Result setsysGetFatalDirtyFlags(u64 *flags_0, u64 *flags_1) {
 
     return rc;
 }
+
+Result setsysGetDeviceNickname(char* buffer) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    ipcAddRecvBuffer(&c, buffer, 0x80, BufferType_Normal);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 77;
+
+    Result rc = serviceIpcDispatch(&g_setsysSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        ipcParse(&r);
+
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+    }
+
+    return rc;
+}
+
+Result setsysSetDeviceNickname(char* buffer) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    ipcAddSendBuffer(&c, buffer, 0x80, BufferType_Normal);
```

Comment:
Hardcoded size for a string sounds fishy to me. Should this be `strlen(buffer)`? If it really needs to be 0x80 bytes in size, then a stack buffer with 0x80 bytes (initialized to zero through `= {0};`) should be used, and the string copied to it.

Reply:
It doesn't seem to matter: I initially put 0x80 since SwIPC implied it needed that. However, when I passed it a 1 megabyte (1024 * 1024 bytes) buffer filled with 'A' and a null terminator at the end (after changing the above ipcAddSendBuffer to use strlen + 1), it didn't crash or return an error, so it seems to just truncate it.